src/core/ext/filters/client_channel/client_channel.cc

```diff
@@ -1268,6 +1273,51 @@ static void resume_pending_batch_in_call_combiner(void* arg,
   grpc_subchannel_call_process_op(subchannel_call, batch);
 }
 
+// The callback to intercept trailing metadata if retries is not enabled
+static void recv_trailing_metadata_ready_for_lb(void* arg, grpc_error* error) {
+  subchannel_batch_data* batch_data = static_cast<subchannel_batch_data*>(arg);
+  grpc_call_element* elem = batch_data->elem;
+  call_data* calld = static_cast<call_data*>(elem->call_data);
+  GPR_ASSERT(calld->pick.recv_trailing_metadata_ready != nullptr);
+  GPR_ASSERT(calld->pick.recv_trailing_metadata != nullptr);
+
+  GRPC_CLOSURE_SCHED(
+      calld->pick.recv_trailing_metadata_ready,
+      GRPC_ERROR_REF(error));
+  calld->pick.recv_trailing_metadata = nullptr;
+  calld->pick.recv_trailing_metadata_ready = nullptr;
+
+  GRPC_CLOSURE_RUN(
+      calld->before_lb_intercept_recv_trailing_metadata_ready,
+      GRPC_ERROR_REF(error));
+}
+
+// Installs a interceptor to inform the lb of the trailing metadata, if needed
+static void maybe_intercept_trailing_metadata_for_lb(
+    void* arg, grpc_transport_stream_op_batch* batch) {
+  subchannel_batch_data* batch_data = static_cast<subchannel_batch_data*>(arg);
+  grpc_call_element* elem = batch_data->elem;
+  call_data* calld = static_cast<call_data*>(elem->call_data);
+  if (calld->pick.recv_trailing_metadata_ready != nullptr) {
+    GPR_ASSERT(calld->pick.recv_trailing_metadata != nullptr);
```

Comment:
I think we should not require this to be non-null, since they might be use-cases where the LB policy wants to know when the call is complete but doesn't actually need to see the trailing metadata.  In that case, it might request a callback but not require us to set the metadata.